test/gtest/possible-contents.cpp

```diff
@@ -810,6 +809,21 @@ TEST_F(PossibleContentsTest, TestStructCones) {
   assertIntersection(
     literalNullA, PossibleContents::fullConeType(funcref), none);
 
+  // Computing intersections is also supported with a Literal.
+  assertIntersection(i32Zero, i32Zero, i32Zero);
+  assertIntersection(i32One, i32Zero, none);
+  assertIntersection(i32Global1, i32Zero, i32Zero);
```

Comment:
Why is zero the intersection of a global and zero?

Reply:
Basically since we consider literal values the most precise thing possible. But we could in theory look at it another way. If we compare a Global to the intersection of a Global and Zero, then if the intersection was a Global we could evaluate it to true, but not if we had a Zero there - not unless we knew the Global contained a Zero. But I think if we do that intersection then we should be able to infer that about the Global in general.

Practically, I think it's useful to keep literals as the most precise/refined things possible.